src/som/interpreter/nodes/literals/BlockNode.java

```diff
@@ -66,7 +66,7 @@ public SInvokable getBlockMethod() {
 
   @Override
   public SBlock executeSBlock(final VirtualFrame frame) {
-    return new SBlock(blockMethod, null, blockClass);
+    return new SBlock(blockMethod, frame.materialize(), blockClass);
```

Comment:
The whole point of this class is to not require the frame.
So, this is wrong here.
Instead, you should make sure that [`requiresContext()`](https://github.com/smarr/SOMns/blob/dev/src/som/compiler/MethodBuilder.java#L223) returns `true`, when the frame is needed